### DISCUSSION TITLE
Link PCAN(Win10) to vcan(WSL2)

### DISCUSSION THREAD

**Discussion by AndyNeubacher** [dorssel/usbipd-win#956] [General]
Hi,

don't know if this is the correct discussion-group here ... but give it a try:

`Q: can i link a physical device in windows (PEAK-USB canbus interface) to a virtual canbus in WSL2 (vcan0)?
`

**reason:**
i'm using a PEAK-USB to connect a Windows-FlashTool to an external device. Usually the Windows-FlashTool communicates over the canbus with an ECU -> but i need that canbus-data to be sent via a UDP-tunnel (WSL2) to a remote ECU

My idea is to link the physical PEAK-USB interface on windows to the virtual-canbus in WSL2. Afterwards the can-data is tunneled in wsl2 via a UDP-socket to the remote device

.... if you have any (other) ideas how to get this link .... any ideas welcome :)

thx &br,
Andy

**Comment by dorssel** [comment]
`usbipd-win` forwards entire USB devices. Once attached to some other host, the USB device is not usable by Windows until you detach it again. Sounds like your USB device actually provides the canbus. As soon as you attach the USB device to WSL2, the canbus is then exposed to WSL, but Windows can no longer see it. Windows-FlashTool won't find any canbus on the Windows host in that case. If all if this is correct about your USB device, it sounds to me like it isn't possible. You would need some sort of canbus forwarder instead.

**Comment by AndyNeubacher** [comment]
hi dorssel,

thx for your immediate response ... unfortunately this is exact the behaviour i'm confronted with :|  ... seems like i need to focus on a different aproach!

thx & br,
Andy
